### ISSUE TITLE
Problem matching LoTW record containing "DATA" as mode

### ISSUE THREAD

**Issue by lu1aax** [magicbug/Cloudlog#2438]
Hi, I am experiencing a problem with confirmation of JS8 qso. Record from LoTW comes with DATA as data mode and that seems to be the reason of not matching.

**Comment by magicbug** [comment]
If you can provide a screenshot of a JS8 confirmed on LOtw website we can probably get to the bottom of it.

**Comment by lu1aax** [comment]
Station
Call Sign	 	LU1AAX
DXCC	 	ARGENTINA (100)
CQ Zone	 	13
ITU Zone	 	14
Grid	 	GF05SJ
Worked Station
Worked	 	K4NXS
DXCC	 	UNITED STATES OF AMERICA (291)
CQ Zone	 	05
ITU Zone	 	08
Grid	 	EM95AC
State	 	South Carolina (SC)
County	 	Spartanburg
Date/Time	 	2023-08-26 20:11:40
Mode	 	DATA (DATA)
Band	 	10M
Frequency	 	28.07850
QSL	 	[2023-08-26 20:29:15](https://lotw.arrl.org/lotwuser/qsodetail?qso=1756870301)
Record ID 1756878972 Received: 2023-08-26 20:29:15

**Comment by int2001** [comment]
JS8 is a "Submode" of MFSK, not of "DATA". That's the reason.

**Comment by phl0** [comment]
The question is which mode and/or submode is contained in the Report that CL downloads from LoTW?

**Comment by int2001** [comment]
@lu1aax can you Post two screenshots (as picture).

1.) from the lotw-page. QSL-Details
2.) out of Cloudlog. QSO-Details

And, if possible: an adif-export of only this QSO from Cloudlog

**Comment by lu1aax** [comment]
Sure

![image](https://github.com/magicbug/Cloudlog/assets/71468941/a2d78358-da7d-48a9-a730-521ca5d93586)

![image](https://github.com/magicbug/Cloudlog/assets/71468941/b9847ff3-f3eb-4f48-88e1-9ee967f609f6)

[LU1AAX-27082023-1221.adi.txt](https://github.com/magicbug/Cloudlog/files/12447948/LU1AAX-27082023-1221.adi.txt)

**Comment by lu1aax** [comment]
> The question is which mode and/or submode is contained in the Report that CL downloads from LoTW?

![image](https://github.com/magicbug/Cloudlog/assets/71468941/dda117ef-4d4c-4f04-a536-7291847d26bf)

**Comment by phl0** [comment]
I just created a test QSO in my CL dev environment. I chose JS8 as mode and then exported an ADIF file. This contains:
```
<MODE:4>MFSK
<SUBMODE:3>JS8
```
which I believe is correct. 
It seems to me that your QSO partner uploaded with different settings which result in DATA/DATA so the modes do not match on import. How do we know that DATA/DATA is JS8 and not something else? I guess K4NXS should check his log software.

**Comment by lu1aax** [comment]
> I just created a test QSO in my CL dev environment. I chose JS8 as mode and then exported an ADIF file. This contains:
> 
> ```
> <MODE:4>MFSK
> <SUBMODE:3>JS8
> ```
> 
> which I believe is correct. It seems to me that your QSO partner uploaded with different settings which result in DATA/DATA so the modes do not match on import. How do we know that DATA/DATA is JS8 and not something else? I guess K4NXS should check his log software.

Well, this was correctly matched in QRZ.com and LoTW, I would expect Cloudlog to do so...

**Comment by phl0** [comment]
I am not sure what the issue is here. AFAIK LoTW does not take modes into calculation for matches because my previous cross-mode QSOs (e.g. me working CW / DX answering in SSB) did match. The LoTW report contains the mode that I sent during my LoTW upload. So it still would be nice to see what the lotwreport.adi contains when you download it from your LoTW account.

**Comment by lu1aax** [comment]
> I am not sure what the issue is here. AFAIK LoTW does not take modes into calculation for matches because my previous cross-mode QSOs (e.g. me working CW / DX answering in SSB) did match. The LoTW report contains the mode that I sent during my LoTW upload. So it still would be nice to see what the lotwreport.adi contains when you download it from your LoTW account.

I've attached the .adi file from LoTW as requested some days ago.

**Comment by phl0** [comment]
Hmm strange. And you uploaded from your Cloudlog to LoTW directly?

**Comment by lu1aax** [comment]
> Hmm strange. And you uploaded from your Cloudlog to LoTW directly?

Yes, I never upload to LoTW with other tools other than Cloudlog.
